# Project Exhebition -1

## Predicting the probability of having Lung Cancer 

## Steps:

**_IMPORTING THE LIBRARIES_**

In [157]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

**_IMPORTING THE DATASET_**

In [158]:
dataset = pd.read_csv('LungCancerSurvey.csv')
x= dataset.iloc[ : , :-1]
y= dataset.iloc[ : , -1]

## Encoding the data 

***ENCODING THE FEATURES***

In [159]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder

ct = ColumnTransformer(transformers = [('encoder' , OneHotEncoder() , [0])], remainder = 'passthrough')
x = np.array(ct.fit_transform(x))


***ENCODING THE DEPENDENT VARIABLE***

In [160]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
y = le.fit_transform(y)

***SPLITTING THE DATASET INTO TRAINING AND TEST SET***

In [161]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=0)

***PRINTING THE NEW SETS***

In [162]:
print(x_train)

[[ 1.  0. 51. ...  2.  2.  1.]
 [ 0.  1. 72. ...  2.  2.  2.]
 [ 0.  1. 59. ...  2.  1.  1.]
 ...
 [ 1.  0. 51. ...  2.  2.  1.]
 [ 1.  0. 64. ...  1.  2.  2.]
 [ 0.  1. 56. ...  1.  1.  1.]]


In [163]:
print(x_test)

[[ 1.  0. 68.  1.  1.  2.  1.  2.  1.  2.  2.  2.  1.  1.  2.  1.]
 [ 0.  1. 64.  2.  2.  2.  2.  1.  2.  2.  1.  2.  2.  2.  1.  2.]
 [ 0.  1. 62.  1.  1.  1.  1.  2.  1.  2.  1.  2.  2.  2.  2.  2.]
 [ 0.  1. 68.  1.  1.  2.  2.  2.  1.  1.  1.  2.  1.  1.  1.  1.]
 [ 1.  0. 67.  2.  2.  2.  2.  1.  2.  1.  1.  1.  1.  1.  1.  1.]
 [ 1.  0. 65.  2.  2.  2.  2.  1.  2.  1.  2.  1.  2.  2.  2.  1.]
 [ 1.  0. 55.  2.  2.  2.  2.  2.  2.  1.  2.  1.  2.  2.  2.  1.]
 [ 1.  0. 71.  2.  2.  2.  2.  2.  2.  1.  2.  1.  2.  1.  2.  2.]
 [ 1.  0. 61.  2.  2.  2.  1.  1.  2.  2.  1.  2.  1.  2.  2.  2.]
 [ 1.  0. 56.  1.  2.  1.  1.  2.  2.  2.  2.  1.  2.  2.  1.  2.]
 [ 0.  1. 64.  1.  2.  2.  2.  1.  1.  2.  1.  2.  1.  1.  2.  2.]
 [ 1.  0. 62.  1.  1.  1.  2.  1.  1.  1.  2.  2.  1.  1.  2.  2.]
 [ 1.  0. 56.  2.  2.  2.  1.  1.  2.  2.  1.  1.  1.  2.  1.  2.]
 [ 0.  1. 52.  1.  2.  2.  1.  2.  1.  2.  2.  2.  2.  1.  2.  1.]
 [ 1.  0. 21.  2.  1.  1.  1.  2.  2.  2.  1.  1.  1.  2.  1. 

In [164]:
print(y_train)

[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 0 1 1 1 0 1 1 0 1 1 0 1 1 1 1 1
 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 0 1 1 0 1 0 1 1 1 1 0 1 1 1 0 0 1 1 0 1 1 1 0 1 1 1 0 1 0 1
 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 0 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1]


In [165]:
print(y_test)

[1 1 1 0 1 1 1 1 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 0 1 1 1 0 1 0 1 1 1 1 1 0 1 1 1 1]


***FITTING THE MACHINE LEARNING MODEL IN THE DATASET TO TRAIN IT***

In [167]:
from sklearn.ensemble import RandomForestClassifier
classifier = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 0)
classifier.fit(x_train, y_train)

RandomForestClassifier(criterion='entropy', n_estimators=10, random_state=0)

***PRINTING THE PREDICTED AND ORIGINAL RESULTS SIDE BY SIDE TO COMPARE***

In [168]:
y_pred = classifier.predict(x_test)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

[[1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [0 1]
 [0 0]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [1 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]]


***PRINTING THE CONFUSION MATRIX AND THE ACCURACY OF THE MODEL***

In [169]:
from sklearn.metrics import confusion_matrix,accuracy_score
print(confusion_matrix(y_test,y_pred))
accuracy_score(y_test,y_pred)

[[ 9  1]
 [ 1 51]]


0.967741935483871

***OBSERVATION : THE ACCURACY OF OUR MACHINE LEARNING MODEL IS 96.774 % WHICH SHOWS THAT OUR MODEL IS WELL ADAPTED TO THE DATASET AND IS ABLE TO PREDICT THE RESULTS WITH HIGH ACCURACY***